src/network/networkd-dhcp6.c

```diff
@@ -112,28 +112,24 @@ static int dhcp6_get_preferred_delegated_prefix(
                 log_link_debug(link, "The requested prefix %s is available. Using it.",
                                strnull(assigned_buf));
                 return 0;
```

Comment:
Right, there's `return 0` so dropping the `else` makes no functional difference.